test/OrchardCore.Tests/Modules/OrchardCore.Tenants/Controllers/AdminControllerTests.cs

```diff
@@ -0,0 +1,247 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.Localization;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using OrchardCore.Data;
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.Notify;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Models;
+using OrchardCore.Recipes.Services;
+using OrchardCore.Settings;
+using OrchardCore.Tenants.Controllers;
+using OrchardCore.Tenants.Services;
+using OrchardCore.Tenants.ViewModels;
+using OrchardCore.Tests.Utilities;
+using Xunit;
+
+namespace OrchardCore.Modules.Tenants.Controllers.Tests
+{
+    public class AdminControllerTests
+    {
+        private readonly IList<ShellSettings> _shellSettings = new List<ShellSettings>();
+        private readonly Mock<IShellHost> _shellHostMock;
+        private readonly Mock<IFeatureProfilesService> _featureProfilesServiceMock;
+
+        public AdminControllerTests()
+        {
+            SeedTenants();
+
+            _shellHostMock = new Mock<IShellHost>();
+            _shellHostMock.Setup(h => h.GetAllSettings()).Returns(_shellSettings);
+            _shellHostMock.Setup(h => h
+                .UpdateShellSettingsAsync(It.IsAny<ShellSettings>()))
+                .Callback<ShellSettings>(s => _shellSettings.Add(s));
+
+            _featureProfilesServiceMock = new Mock<IFeatureProfilesService>();
+            _featureProfilesServiceMock.Setup(fp => fp.GetFeatureProfilesAsync())
+                .Returns(Task.FromResult((IDictionary<string, FeatureProfile>)new Dictionary<string, FeatureProfile>
+                {
+                    { "Feature Profile", new FeatureProfile() }
+                }));
+        }
+
+        [Fact]
+        public async Task CreateTenant()
+        {
+            // Arrange
```

Comment:
These are unnecessary comments.

Reply:
We already used AAA pattern in our testing

Reply:
We already did that in almost if not all the tests I saw

Reply:
You're right, a lot of them do have it. While I disagree with the practice, this is not the right time to debate that in general, so let's leave these comments.